provider/test_utils.go

```diff
@@ -75,3 +77,29 @@ func TestCheckResourceAttrNot(name, key, value string) resource.TestCheckFunc {
 		return nil
 	}
 }
+
+func keycloakVersionIsGreaterThanOrEqualTo(keycloakMajorVersion int) (bool, error) {
+	keycloakVersion := os.Getenv("KEYCLOAK_VERSION")
+	if len(keycloakVersion) > 0 {
+		i, err := strconv.Atoi(string(keycloakVersion[0]))
+		if err == nil {
+			return i >= keycloakMajorVersion, nil
+		}
+	}
+	return false, fmt.Errorf("KEYCLOAK_VERSION env var was not correctly set, it was '%s'. It should be for example : 5.0.0, 10.0.1", keycloakVersion)
+}
+
+func keycloakServerInfoVersionIsGreaterThanOrEqualTo(keycloakClient *keycloak.KeycloakClient, keycloakMajorVersion int) (bool, error) {
+	serverInfo, err := keycloakClient.GetServerInfo()
+	if err != nil {
+		return false, fmt.Errorf("/serverInfo endpoint retuned an error, server Keycloak version could not be determined")
+	}
+	keycloakVersion := serverInfo.SystemInfo.ServerVersion
+	if len(keycloakVersion) > 0 {
+		i, err := strconv.Atoi(string(keycloakVersion[0]))
+		if err == nil {
+			return i >= keycloakMajorVersion, nil
+		}
+	}
```

Comment:
This breaks when  ` version >= 10`.

What do you think of making use of hashicorp's [go-version](https://github.com/hashicorp/go-version) library?  It'd be nice to be able to target specific minor/patch versions as well.

Reply:
Good point, and i used the go-version lib, thanks for the tip.
I am learning new things